File: 0230-kth-smallest-element-in-a-bst/0230-kth-smallest-element-in-a-bst.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def kthSmallest(self, root: Optional[TreeNode], k: int) -> int:
        res = [0, 0]
        def check(node):
            if res[1]:
                return
            if not node:
                return
            check(node.left)
            res[0] += 1
            if res[0] == k:
                res[1] = node.val
            check(node.right)
        check(root)
        return res[1]

File: 0230-kth-smallest-element-in-a-bst/0230-kth-smallest-element-in-a-bst.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @param {number} k
 * @return {number}
 */
var kthSmallest = function(root, k) {
    let count = 0;
    let res = 0;
    const check = (node) => {
        if (res) return;
        if (!node) return;
        check(node.left);
        count++;
        if (count === k) res = node.val;
        check(node.right);
    }
    check(root);
    return res;
};